server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java

```diff
@@ -22,7 +32,43 @@ public StrandService(StrandRepository strandRepository) {
 
   public StrandModel createStrand(Request request) {
     StrandModel strand =
-        new StrandModel(request.getName(), request.getIpAddress(), request.getPort());
+        StrandModel.builder()
+            .name(request.getName())
+            .ipAddress(request.getIpAddress())
+            .port(request.getPort())
+            .build();
     return strandRepository.save(strand);
   }
+
+  // Run strands check every 10 minutes
+  @Scheduled(fixedDelayString = "${schedule.strandAvailabilityMS}")
+  public void checkStrandsAvailability() {
+    logger.info("Starting Strand availability cron job.");
+    List<StrandModel> strands = strandRepository.findAll();
+    strands.forEach(
+        strand -> {
+          try {
+            logger.info("Getting health status of " + strand.getFullAddress());
+            ResponseEntity<String> response =
+                restTemplate.getForEntity(
+                    "https://" + strand.getFullAddress() + "/health", String.class);
+            if (response.getStatusCode().is2xxSuccessful()) {
+              strand.setAvailable(true);
+            } else {
+              logger.error(
+                  "Strand " + strand.getName() + " failed the health check with a non-200 status.");
+              strand.setAvailable(false);
+            }
+          } catch (RestClientException ex) {
+            logger.error(
+                "Strand "
+                    + strand.getName()
+                    + " failed the health check with the following error: "
+                    + ex.getMessage());
```

Comment:
Oh man this makes me miss formatted strings in JS and Python

Reply:
Tell me about it